RecoTracker/MkFit/plugins/MkFitGeometryESProducer.cc

```diff
@@ -262,6 +278,36 @@ void MkFitGeometryESProducer::fillShapeAndPlacement(const GeomDet *det,
   layer_info.set_subdet(detid.subdetId());
   layer_info.set_is_pixel(detid.subdetId() <= 2);
   layer_info.set_is_stereo(trackerTopo_->isStereo(detid));
+
+  bool doubleSide = false;  //double modules have double material
+  if (detid.subdetId() == SiStripSubdetector::TIB)
+    doubleSide = trackerTopo_->tibIsDoubleSide(detid);
+  else if (detid.subdetId() == SiStripSubdetector::TID)
+    doubleSide = trackerTopo_->tidIsDoubleSide(detid);
+  else if (detid.subdetId() == SiStripSubdetector::TOB)
+    doubleSide = trackerTopo_->tobIsDoubleSide(detid);
+  else if (detid.subdetId() == SiStripSubdetector::TEC)
+    doubleSide = trackerTopo_->tecIsDoubleSide(detid);
+
+  if (!doubleSide)  //fill material
+  {
+    //module material
+    float bbxi = det->surface().mediumProperties().xi();
+    float radL = det->surface().mediumProperties().radLen();
+    //loop over bins to fill histogram with bbxi, radL and their weight, which the overlap surface in r-z with the cmsquare of a bin
+    for (unsigned int i = 0; i < mkfit::Config::nBinsZMat; i++) {
+      for (unsigned int j = 0; j < mkfit::Config::nBinsRMat; j++) {
+        constexpr float iBin = mkfit::Config::nBinsZMat / mkfit::Config::rangeZMat;
+        constexpr float jBin = mkfit::Config::nBinsRMat / mkfit::Config::rangeRMat;
+        float iF = i * iBin;
+        float jF = j * jBin;
+        float overlap = std::max(0.f, std::min(jF + 1, rbox_max) - std::max(jF, rbox_min)) *
+                        std::max(0.f, std::min(iF + 1, zbox_max) - std::max(iF, zbox_min));
```

Comment:
on a second look, I'm not sure this should be `jF + 1` or `jF + jBin` instead.
Can the loop be effectively written to be from `std::floor(zbox_min/iBin)` to `std::ceil(zbox_max/iBin)`, and similarly for `rbox`?
This would make it really faster.

Reply:
iBin jBin is correct, but I thought it wouldn't matter even if we decrease the bin size, so I just didn't update it.

for the loop, the speedup should is a good idea, and I did it for the fillLayers loops already, so I am going to wait for the review and change it while addressing other comments

Reply:
so, the code should be  like this?
```suggestion
                        std::max(0.f, std::min(iF + iBin, zbox_max) - std::max(iF, zbox_min));
```

Better correct this.

Reply:
> for the loop, the speedup should is a good idea, and I did it for the fillLayers loops already, so I am going to wait for the review and change it while addressing other comments

with my past reco hat on, I'd be more happy to see no additional updates on top of what's in the formal cmssw repo review.
Since these comments are just in trackreco, more changes would seem to be unexpected there.